add missing or correct errors in header guards for 9 files

fwph opened the pull request:
# Description

The code changes in this PR were generated automatically by the Permanence AI Coder and reviewed by myself. This bugfix PR updates several header files to add include guards which were previously missing and two files to fix include guards which were incorrect.

        - modify sources/core-sw/src/compression/opt/placeholder.h (add missing include guard)
        - modify sources/core-sw/src/filtering/opt/qplc_unpack_16u_k0.h (add missing include guard)
        - modify sources/core-sw/src/filtering/opt/qplc_unpack_8u_k0.h (add missing include guard)
        - modify sources/core-sw/src/filtering/opt/qplc_unpack_be_16u_k0.h (add missing include guard)
        - modify sources/core-sw/src/filtering/opt/qplc_unpack_be_32u_k0.h (add missing include guard)
        - modify sources/core-sw/src/filtering/opt/qplc_unpack_be_8u_k0.h (add missing include guard)
        - modify sources/isal/igzip/huffman.h (add missing include guard)
        - modify sources/isal/igzip/igzip_wrapper.h (add missing #define inside include guard)
        - modify sources/isal/igzip/static_inflate.h (incorrectly located #endif of include guard)

on-behalf-of: @permanence-ai github-ai@permanence.ai

# Checklist

- No new tests as this is purely a build-time correctness update
- Software path test results had only expected failures (per latest release notes) following the changes:

```
[----------] Global test environment tear-down
[==========] 385 tests from 125 test suites ran. (441480 ms total)
[  PASSED  ] 334 tests.
[  SKIPPED ] 45 tests, listed below:
. . .
[  FAILED  ] 6 tests, listed below:
[  FAILED  ] ta_c_api_deflate_stateful.dynamic_default_verify
[  FAILED  ] ta_c_api_deflate_stateful.dynamic_high_verify
[  FAILED  ] ta_c_api_deflate_stateful.fixed_default_verify
[  FAILED  ] ta_c_api_deflate_stateful.fixed_high_verify
[  FAILED  ] ta_c_api_deflate_stateful.static_default_verify
[  FAILED  ] ta_c_api_deflate_stateful.static_high_verify
```

fwph commented:
note: https://github.com/intel/qpl/pull/39 is a superset of this PR

mzhukova commented on sources/core-sw/src/compression/opt/placeholder.h:
```diff
@@ -4,7 +4,12 @@
  * SPDX-License-Identifier: MIT
  ******************************************************************************/
 
+#ifndef PLACEHOLDER_H
+#define PLACEHOLDER_H
```
I understand that QPL doesn't follow it everywhere, but in order to improve, could we follow the convention `QPL_<dir_name_1>_<dir_name_2>_..._<dirN>_<header_file_name>_H/HPP`?

fwph replied:
no problem. will add a config policy around this and regenerate the branch.

mzhukova reviewed (changes requested):
I would like style question to be resolved before approving.

Permanence-AI-Coder committed:
add missing or correct errors in header guards for 9 files

    - modify sources/core-sw/src/compression/opt/placeholder.h (add missing include guard)
    - modify sources/core-sw/src/filtering/opt/qplc_unpack_16u_k0.h (add missing include guard)
    - modify sources/core-sw/src/filtering/opt/qplc_unpack_8u_k0.h (add missing include guard)
    - modify sources/core-sw/src/filtering/opt/qplc_unpack_be_16u_k0.h (add missing include guard)
    - modify sources/core-sw/src/filtering/opt/qplc_unpack_be_32u_k0.h (add missing include guard)
    - modify sources/core-sw/src/filtering/opt/qplc_unpack_be_8u_k0.h (add missing include guard)
    - modify sources/isal/igzip/huffman.h (add missing include guard)
    - modify sources/isal/igzip/igzip_wrapper.h (add missing #define inside include guard)
    - modify sources/isal/igzip/static_inflate.h (incorrectly located #endif of include guard)

on-behalf-of: @permanence-ai <github-ai@permanence.ai>

fwph commented:
I've updated this PR as part of regenerating the branch for  #39 -- 39 is still a superset including the same change _and_ the same commit.

mzhukova commented:
> I've updated this PR as part of regenerating the branch for #39 -- 39 is still a superset including the same change _and_ the same commit.

Understood on #39, however I'd like to keep changes small and self-contained, therefore it makes sense to introduce this one separately and rebase the rest on top of it.
Also, just a note, per [Contributing Guide](https://github.com/intel/qpl/blob/develop/CONTRIBUTING.md#contribution-flow), I'll have to cherry-pick this changes and do internal testing prior merging.

fwph commented:
Makes sense to keep them separate! We will keep an eye out here if there are any other changes for this PR. Thanks for reviewing.

fwph commented:
hi @mzhukova -- wanted to check on the status of this PR and if it's entered internal testing. thanks!

mzhukova commented:
Hi @fwph, sorry for the delayed response. 
The changes from this PR are merged now, so closing this one.

fwph commented:
Thanks!
